src/main/java/com/r307/arbitrader/service/NotificationServiceImpl.java

```diff
@@ -25,56 +27,29 @@
 
     private final JavaMailSender javaMailSender;
     private final NotificationConfiguration notificationConfiguration;
+    private final TelegramClient telegramClient;
 
     @Inject
-    public NotificationServiceImpl(JavaMailSender javaMailSender, NotificationConfiguration notificationConfiguration) {
+    public NotificationServiceImpl(JavaMailSender javaMailSender, NotificationConfiguration notificationConfiguration, TelegramClient telegramClient) {
         this.javaMailSender = javaMailSender;
         this.notificationConfiguration = notificationConfiguration;
+        this.telegramClient = telegramClient;
     }
 
     /**
-     * Send an email notification, if email is configured.
-     *
-     * @param subject The subject line of the email.
-     * @param body The body of the email.
-     */
-    @Override
-    public void sendEmailNotification(String subject, String body) {
-        if (notificationConfiguration.getMail() == null || notificationConfiguration.getMail().getActive() == null ||
-            !notificationConfiguration.getMail().getActive()) {
-
-            LOGGER.info("Email notification is disabled");
-            return;
-        }
-
-        SimpleMailMessage mail = new SimpleMailMessage();
-        mail.setTo(notificationConfiguration.getMail().getTo());
-        mail.setFrom(notificationConfiguration.getMail().getFrom());
-        mail.setSubject(subject);
-        mail.setText(body);
-
-        try {
-            javaMailSender.send(mail);
-        }
-        catch (Exception e) {
-            LOGGER.error("Could not send email notification to {}. Reason: {}", notificationConfiguration.getMail().getTo(), e.getMessage());
-        }
-    }
-
-    /**
-     * Formats and sends an email when a trade is entered.
+     * Formats and sends a notification when we trade an entry position.
      *
      * @param spread The Spread.
      * @param exitTarget The exit target.
      * @param longVolume The long exchange volume.
      * @param longLimitPrice The long exchange limit price.
      * @param shortVolume The short exchange volume.
      * @param shortLimitPrice The short exchange limit price.
+     * @param isForceEntryPosition Flag to indicate if this entry trade is forced or not.
      */
     @Override
-    public void sendEmailNotificationBodyForEntryTrade(Spread spread, BigDecimal exitTarget, BigDecimal longVolume,
-                                                           BigDecimal longLimitPrice, BigDecimal shortVolume,
-                                                           BigDecimal shortLimitPrice) {
+    public void sendEntryTradeNotification(Spread spread, BigDecimal exitTarget, BigDecimal longVolume, BigDecimal longLimitPrice,
```

Comment:
I like that this is getting more generic. Nice job. 👍

Reply:
Yeah, looks much better now :)